src/modal/modal.jsx

```diff
@@ -0,0 +1,65 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Modal } from '@mui/material';
+
+export function PSAModal({
+  modalContent, 
+  sx, 
+  style, 
+  open, 
+  onClose, 
+  closeAfterTransition, 
+  disableEscapeKeyDown
```

Comment:
Can you do the same thing as you did in tooltip to remove some props and use `...props` instead?